numba/cuda/tests/cudadrv/test_deallocations.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+
+from numba import cuda, config
+from numba.cuda.testing import unittest, skip_on_cudasim
+
+
+@skip_on_cudasim('not supported on CUDASIM')
+class TestDeallocation(unittest.TestCase):
+    def test_max_pending_count(self):
+        # get deallocation manager and flush it
+        deallocs = cuda.current_context().deallocations
+        deallocs.clear()
+        self.assertEqual(len(deallocs), 0)
+        # deallocate to maximum count
+        for i in range(config.CUDA_DEALLOCS_COUNT):
+            cuda.to_device(np.arange(1))
+            self.assertEqual(len(deallocs), i + 1)
+        # one more to trigger .clear()
+        cuda.to_device(np.arange(1))
+        self.assertEqual(len(deallocs), 0)
+
+    def test_max_pending_bytes(self):
+        # get deallocation manager and flush it
+        deallocs = cuda.current_context().deallocations
+        deallocs.clear()
+        self.assertEqual(len(deallocs), 0)
+
+        max_pending = deallocs._max_pending_bytes
```

Comment:
I'm worried this test may eat a lot of memory even on the host (if you have a large GPU). Can't you change `max_pending_bytes` for the during of the test?